File: app/src/main/java/com/qiaoqiao/repository/backend/Wikipedia.java
package com.qiaoqiao.repository.backend;

import android.support.annotation.NonNull;

import com.qiaoqiao.repository.backend.model.wikipedia.WikiResult;
import com.qiaoqiao.repository.backend.model.wikipedia.geo.GeoResult;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Url;

public interface Wikipedia {
	final class WikiReqBody {
		@NonNull String language;
		@NonNull String keyword;

		public WikiReqBody(@NonNull String language, @NonNull String keyword) {
			this.language = language;
			this.keyword = keyword;
		}
	}

	@POST("documents/wikipedia")
	Observable<WikiResult> getResult(@Body WikiReqBody body);

	@GET
	Observable<GeoResult> getGeosearch(@Url String url);
}

File: app/src/main/java/com/qiaoqiao/repository/backend/Wikipedia.kt
package com.qiaoqiao.repository.backend

import com.qiaoqiao.repository.backend.model.wikipedia.WikiResult
import com.qiaoqiao.repository.backend.model.wikipedia.geo.GeoResult
import io.reactivex.Flowable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Url

interface Wikipedia {

    @POST("documents/wikipedia")
    fun getResult(@Body body: KnowledgeRequest): Flowable<WikiResult>

    @GET
    fun getResult(@Url queryWithPageId: String): Flowable<WikiResult>

    @GET
    fun getGeosearch(@Url url: String): Flowable<GeoResult>

}
